src/Shape.jl

```diff
@@ -0,0 +1,275 @@
+struct Shape
+    points::Vector{Point}
+    simplified_points::Vector{Point}
+    centroid::Point
+    centered_points::Vector{Point}
+    subpaths::Vector{Vector{Point}}
+    num_acute_angles::Int
+    num_obtuse_angles::Int
+    num_right_angles::Int
+end
+
```

Comment:
Everything here is great, but to be honest, I have no idea what the general purpose of these functions are. Could you add small dev docstrings to them so I could better review and understand what each one is doing? Thanks!